CONTRIBUTING.md

```diff
@@ -32,6 +32,50 @@ The comfort and safety of Layer5 community members are our priority. Please do w
 
 To help you get started on contributing to Layer5 projects, refer to the [Layer5 Newcomers Guide](https://layer5.io/community/newcomers), which aims to make contributions easier for new folks like you! See the resources and tutorials to help you get started. Along with this we have an extensive handbook on how this community works , how folks inside the community wear different hats to run it and few of its guidelines, have a look inside the [Community Handbook](https://layer5.io/community/handbook).
 
+## Issues & Pull Requests
+
+### Creating an Issue
+
+Before **creating** an Issue i.e for `features`/`bugs`/`improvements` please follow these steps:
+
+
+1. search existing Issues before creating a new issue (look to see if the issue has already been created)
+1. if it doesn't exist create a new issue giving as much context as possible (please take note and select the correct Issue type, for example `bug`, `documentation` or `feature`)
+1. if you wish to work on the Issue once it has been triaged, please include this in your Issue description
+
+### Working on an Issue (get it assigned to you)
+
+Before working on an existing Issue please follow these steps:
+
+1. comment asking for the issue to be assigned to you (do not tag maintainers on GitHub or Slack as all maintainers receive your comment notifications)
+1. after the Issue is assigned to you, you can start working on it
+1. **only** start working on this Issue (and open a Pull Request) when it has been assigned to you - this will prevent confusion, multiple people working on the same issue and work not being used
+1. reference the Issue in your Pull Request (for example `This PR fixes #123`)
+
+> Notes:
+>
+> - check the `Assignees` box at the top of the page to see if the issue has been assigned to someone else before requesting this be assigned to you
+> - if an Issue is unclear, ask questions to get more clarity before asking to have the Issue assigned to you
+> - only request to be assigned an Issue if you know how to work on it
```

Comment:
From my vantage point, this should be point #1 and is where we're experiencing the most pain today.